nestkernel/secondary_event.h

```diff
@@ -48,14 +50,14 @@ class SecondaryEvent : public Event
 
   virtual void add_syn_id( const synindex synid ) = 0;
 
-  virtual bool supports_syn_id( const synindex synid ) const = 0;
+  virtual bool has_synid( const synindex synid ) const = 0;
```

Comment:
It seems that this function is not used anywhere after your change from `vector` to `set`. I suggest to remove it.